File: frontend/src/api/candidates.ts
import client from './client'

export interface Candidate {
  id: string
  name: string
  party: string
  party_short: string
  party_symbol_url: string
  photo_url: string
  serial_number: number
  criminal_cases: CriminalCase[]
  total_assets_inr: number
  assets_detail: AssetsDetail
  education: string
  education_discrepancy: boolean
  affidavit_url: string
}

export interface CriminalCase {
  section: string
  description: string
  court: string
  year: number
}

export interface AssetsDetail {
  movable: number
  immovable: number
  vehicles: number
  bank_balance: number
}

export const getCandidates = async (state: string, constituency: string, year = 2026, constituencyName = ''): Promise<{ candidates: Candidate[]; count: number }> => {
  const params = constituencyName ? { name: constituencyName } : {}
  const { data } = await client.get(`/candidates/${state}/${constituency}/${year}`, { params })
  return data
}

export const getSchedule = async (state: string, constituency: string) => {
  const { data } = await client.get(`/schedule/${state}/${constituency}`)
  return data
}

export const getDocuments = async (state: string) => {
  const { data } = await client.get(`/documents/${state}`)
  return data
}

export const formatAssets = (inr: number): string => {
  if (inr >= 10000000) return `₹${(inr / 10000000).toFixed(1)} Cr`
  if (inr >= 100000)   return `₹${(inr / 100000).toFixed(1)} Lakh`
  return `₹${inr.toLocaleString('en-IN')}`
}

File: backend/routers/candidates.py
from fastapi import APIRouter, HTTPException, Query
from typing import Optional
from services.eci_service import get_candidates, get_schedule, get_documents_for_state

router = APIRouter(prefix="/api", tags=["candidates"])


@router.get("/candidates/{state}/{constituency}/{year}")
async def candidates(
    state: str,
    constituency: str,
    year: int,
    name: Optional[str] = Query(default=None, description="Assembly constituency name for Wikipedia lookup"),
):
    data = await get_candidates(state, constituency, year, constituency_name=name or "")
    return {"candidates": data, "count": len(data)}


@router.get("/schedule/{state}/{constituency}")
async def schedule(state: str, constituency: str):
    data = await get_schedule(state, constituency)
    if not data:
        raise HTTPException(404, "Schedule not found")
    return data


@router.get("/documents/{state}")
async def documents(state: str):
    return get_documents_for_state(state.upper())
